File: solutions/r/resistor-color-duo/2/resistor-color-duo.R
value <- function(resistor_colors) {
  color_vals <- c(black=0, brown=1, red=2, orange=3,
              yellow=4, green=5, blue=6, violet=7,
              grey=8, white=9)
  v1 <- color_vals[resistor_colors[1]]
  v2 <- color_vals[resistor_colors[2]]
  as.integer(10*v1  + v2)
}

File: solutions/free-pascal/resistor-color-duo/1/ResistorColorDuo.pas
unit ResistorColorDuo;

{$mode ObjFPC}{$H+}

interface

type
  TStrArray = Array Of String;

function value(const colors : TStrArray) : int8;

implementation

uses SysUtils;

function colorVal(color : string) : int8;
var 
  colors : TStrArray;
  i : integer;
begin
  colors := ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];
  
  for i:=0 to 9 do
  begin
    if colors[i] = color then
      exit(i);
  end;
end;



function value(const colors : TStrArray) : int8;
var 
  c1, c2 : integer;
begin
  c1 := colorVal(colors[0]);
  c2 := colorVal(colors[1]);
  result := 10*c1 + c2;
end;

end.
